src/partition/repart.c

```diff
@@ -2231,21 +2234,32 @@ static int context_load_partition_table(Context *context) {
         if (!c)
                 return log_oom();
 
-        if (arg_sector_size > 0)
+        if (arg_sector_size > 0) {
                 r = fdisk_save_user_sector_size(c, /* phy= */ 0, arg_sector_size);
-        else {
+                fs_secsz = arg_sector_size;
```

Comment:
doesn't matter much, but we usually say that function calls and their immediate error handling should stay together in our code. Which would suggest to swap these two lines, so that the fdisk_save_user_sector_size() call is last in this block, and thus closest to the error handling that comes after the else branch...